server/src/main/java/org/opensearch/ratelimitting/admissioncontrol/controllers/CPUBasedAdmissionController.java

```diff
@@ -10,46 +10,109 @@
  *  It provides methods to apply admission control if configured limit has been reached
  */
 public class CPUBasedAdmissionController extends AdmissionController {
+    public static final String CPU_BASED_ADMISSION_CONTROLLER = "global_cpu_usage";
     private static final Logger LOGGER = LogManager.getLogger(CPUBasedAdmissionController.class);
     public CPUBasedAdmissionControllerSettings settings;
 
     /**
-     *
-     * @param admissionControllerName State of the admission controller
+     * @param admissionControllerName       Name of the admission controller
+     * @param resourceUsageCollectorService Instance used to get node resource usage stats
+     * @param clusterService                ClusterService Instance
+     * @param settings                      Immutable settings instance
      */
-    public CPUBasedAdmissionController(String admissionControllerName, Settings settings, ClusterSettings clusterSettings) {
-        super(new AtomicLong(0), admissionControllerName);
-        this.settings = new CPUBasedAdmissionControllerSettings(clusterSettings, settings);
+    public CPUBasedAdmissionController(
+        String admissionControllerName,
+        ResourceUsageCollectorService resourceUsageCollectorService,
+        ClusterService clusterService,
+        Settings settings
+    ) {
+        super(admissionControllerName, resourceUsageCollectorService, clusterService);
+        this.settings = new CPUBasedAdmissionControllerSettings(clusterService.getClusterSettings(), settings);
     }
 
     /**
-     * This function will take of applying admission controller based on CPU usage
+     * Apply admission control based on process CPU usage
      * @param action is the transport action
      */
     @Override
-    public void apply(String action) {
-        // TODO Will extend this logic further currently just incrementing rejectionCount
+    public void apply(String action, AdmissionControlActionType admissionControlActionType) {
         if (this.isEnabledForTransportLayer(this.settings.getTransportLayerAdmissionControllerMode())) {
-            this.applyForTransportLayer(action);
+            this.applyForTransportLayer(action, admissionControlActionType);
         }
     }
 
-    private void applyForTransportLayer(String actionName) {
-        // currently incrementing counts to evaluate the controller triggering as expected and using in testing so limiting to 10
-        // TODO will update rejection logic further in next PR's
-        if (this.getRejectionCount() < 10) {
-            this.addRejectionCount(1);
+    /**
+     * Apply transport layer admission control if configured limit has been reached
+     */
+    private void applyForTransportLayer(String actionName, AdmissionControlActionType admissionControlActionType) {
+        if (isLimitsBreached(actionName, admissionControlActionType)) {
+            this.addRejectionCount(admissionControlActionType.getType(), 1);
```

Comment:
This should be in below `if` condition where we are actully rejecting.

Reply:
Actually there is `monitor_only` mode where we will track the rejections but will not do actual rejection. So it was added before and we are only rejecting if the mode is `enforced`.

Reply:
But what if we even didnt specify `monitor_only` mode ? Let say I have just disabled AdmissionControl or I am not using that feature as per my requirement, this will print additional logging/computation in that case.

Reply:
We are checking disabled or not and only proceeding further so there won't be any evaluation and no additional logging.